src/lib.rs

```diff
@@ -936,13 +947,15 @@ impl EGraph {
         let search_start = Instant::now();
         let mut searched = vec![];
         for (name, rule) in copy_rules.iter() {
-            let mut all_values = vec![];
+            let mut all_matches = vec![];
             if rule.banned_until <= iteration {
                 let mut fuel = safe_shl(match_limit, rule.times_banned);
                 let rule_search_start = Instant::now();
+                let mut did_match = false;
                 self.run_query(&rule.query, rule.todo_timestamp, |values| {
+                    did_match = true;
                     assert_eq!(values.len(), rule.query.vars.len());
-                    all_values.extend_from_slice(values);
+                    all_matches.extend_from_slice(values);
```

Comment:
Is `did_match` the same as `all_matches` being non empty?

Reply:
No- since `all_matches` is a flat vector of matches, when there are no variables it is always empty.

Reply:
Cool, that makes sense!